app/assets/javascripts/admin/addon/components/admin-user-fields-form.gjs

```diff
@@ -0,0 +1,273 @@
+import Component from "@glimmer/component";
+import { cached, tracked } from "@glimmer/tracking";
+import { hash } from "@ember/helper";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import { schedule } from "@ember/runloop";
+import { service } from "@ember/service";
+import { eq, or } from "truth-helpers";
+import Form from "discourse/components/form";
+import PluginOutlet from "discourse/components/plugin-outlet";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { i18n } from "discourse-i18n";
+import ValueList from "admin/components/value-list";
+import UserField from "admin/models/user-field";
+
+export default class AdminUserFieldsForm extends Component {
+  @service dialog;
+  @service router;
+  @service adminUserFields;
+  @service adminCustomUserFields;
+
+  @tracked
+  editableDisabled =
+    this.args.userField?.requirement === "for_all_users";
+  userField;
+  originalRequirement;
+
+  constructor() {
+    super(...arguments);
+    this.originalRequirement = this.args.userField.requirement;
+  }
+
+  get fieldTypes() {
+    return UserField.fieldTypes();
+  }
+
+  @cached
+  get formData() {
+    return this.args.userField.getProperties(
+      "field_type",
+      "name",
+      "description",
+      "requirement",
+      "editable",
+      "show_on_profile",
+      "show_on_user_card",
+      "searchable",
+      "options",
+      ...this.adminCustomUserFields.additionalProperties
+    );
+  }
+
+  @action
+  setRequirement(value, { set }) {
+    set("requirement", value);
+
+    if (value === "for_all_users") {
+      this.editableDisabled = true;
+      set("editable", true);
+    } else {
+      this.editableDisabled = false;
+    }
+  }
+
+  @action
+  async save(data) {
+    let confirm = true;
+
+    if (
+      data.requirement === "for_all_users" &&
+      this.originalRequirement !== "for_all_users"
+    ) {
+      confirm = await this._confirmChanges();
+    }
+
+    if (!confirm) {
+      return;
+    }
+
+    try {
+      const isNew = this.args.userField.isNew;
+
+      await this.args.userField.save(data);
+
+      if (this.isDestroying || this.isDestroyed) {
+        return;
+      }
```

Comment:
DO you know why this is necessary?

Reply:
Probably not here at least. Seems like this should either stay in the listing component or be removed. I'll investigate. 👍